src/main/java/org/opensearch/knn/training/TrainingJob.java

```diff
@@ -208,4 +212,13 @@ public void run() {
             nativeMemoryCacheManager.invalidate(modelAnonymousEntryContext.getKey());
         }
     }
+
+    private void setIndexCreatedVersionInKnnMethodContext() {
+        final Version indexCreatedVersion = trainingDataEntryContext.getClusterService()
```

Comment:
Can we just call: `trainingDataEntryContext.getClusterService().localNode().getVersion()`. Training index might be on an older version and it should not influence the parameters set for the model, so I think we just want to use whatever the node's version is.

Reply:
@jmazanec15 
actually no, we should use the version of the index and not the node for consistency, because what a training index represent is the state of attributes for doing the training and not where it is running. 

This is similar to other indices that got created with an older version of Opensearch we using their indexCreatedVersion to identify the defaults.

Reply:
No, I disagree. A training index only contains the data used for training - nothing else. It does not need to even have `index.knn=true` to work. The model's state should not be tied to the training index outside the training data itself.

Reply:
With your suggestion we are tieing the parameters with the node version on which the training is running. Hence I want to tie the parameters default where the data for training is coming in.

Reply:
> With your suggestion we are tieing the parameters with the node version on which the training is running(Not a good design as per my thoughts). Hence I want to tie the parameters default where the data for training is coming in.

Reply:
If we ever need an index parameter for training, I agree with navneet1v that it should come from index version but not from node version.

However, I think jmazanec15 is saying that, currently there is no index parameter that training task needs and it might be true even in the future?

navneet1v, is this code needed or is it just a preparation for future?

Reply:
This code is needed. We need to deicide from which version to pick? Is it the version of node or version of training index. 
@heemin32 , @jmazanec15

Reply:
I think tying to the training index will be worse than node/cluster version where model is created. Assume someone upgrades from 2.11 to 2.12 and they have created their training index on 2.11. If we tie the version to the training index, we would need to explain that the training index needs to be re-indexed in order to get the new defaults. 

The role of training index is supposed to be simple - it is just any index that contains the training vectors you want to train your model with. Tying the state of that index to the model complicates this picture.

Reply:
No actually its other way around. The data of the index represents how the model is getting trained. 

Tying this information with node version will make training inconsistent. 

Example I am on 2.11 version trained a model using training index-1. Now I upgraded the cluster to 2.12 used the same training index to train another model. Now the older model and this model will be different. But they got trained on same data it should be same. 

Tying to node state is making training inconsistent and there is no predictability as node version cannot be same. But with index created version it’s always same doesn’t matter on which version if cluster we are running.

@jmazanec15

Reply:
Okay. Now I got better knowledge around the topic. I am with @jmazanec15 now.
Currently, when we train a model, we don't copy any configuration parameter from source index. In that sense, there is no strong relationship between source index and the training index. Then, the default value should be decided based on node's version(training index version) but not by source index version.

Reply:
After understanding more around how training works, I am with @jmazanec15 now.

Here the training index is source index from which we only use its vector data but nothing more. Currently, when we train a model, we don't copy any configuration parameter from source index. In that sense, there is no strong relationship between source index and the training index. Then, the default value should be decided based on node's version(version of model at the time of creation) but not by source index version.

Reply:
> Here the training index is source index from which we only use its vector data but nothing more.

This was the old understanding, because we never made changes in the default values. 


> Then, the default value should be decided based on node's version(version of model at the time of creation) but not by source index version.

Why node version then? we should decide the based on the highest version that cluster can support. 


Also, I am not able to understand why we are favoring inconsistency in model training here? What is the reason behind not favoring consistency?

Reply:
Conclusion:
Did some more discussion with @vamshin, @jmazanec15 and we concluded that we should the node version here rather than the training index created version.

User experience:
With the above conclusion done, the user experience will be during a mixed cluster if training lands on a node with older version of Opensearch it will use the older defaults, but if training job lands on new version of Opensearch it will use the new defaults.